lib/UI/calendar/widgets/date_widget.dart

```diff
@@ -3,96 +3,97 @@ import 'package:exerlog/Bloc/workout_bloc.dart';
 import 'package:exerlog/Models/workout.dart';
 import 'package:exerlog/UI/global.dart';
 import 'package:exerlog/UI/prev_workout/prev_workout_page.dart';
-import 'package:exerlog/src/widgets/gradient_button.dart';
 import 'package:flutter/material.dart';
 
-class DateWidget extends StatefulWidget {
+class DateWidget extends StatelessWidget {
   final DateTime date;
 
   DateWidget(this.date);
 
-  @override
-  _DateWidgetState createState() => _DateWidgetState();
-}
-
-class _DateWidgetState extends State<DateWidget> {
   @override
   Widget build(BuildContext context) {
-    DateTime after = new DateTime(
-        widget.date.year, widget.date.month, widget.date.day, 23, 59, 59, 0, 0);
-    DateTime before = new DateTime(
-        widget.date.year, widget.date.month, widget.date.day, 0, 0, 0, 0, 0);
+    DateTime after = DateTime(
+      date.year,
+      date.month,
+      date.day,
+      23,
+      59,
+      59,
+      0,
+      0,
+    );
+    DateTime before = DateTime(
+      date.year,
+      date.month,
+      date.day,
+      0,
+      0,
+      0,
+      0,
+      0,
+    );
     return StreamBuilder<QuerySnapshot<Map<String, dynamic>>>(
       stream: getWorkoutOnDate(after, before),
       builder: (context, snapshot) {
-        if (snapshot.connectionState == ConnectionState.waiting) {
-          return Center(
-            child: CircularProgressIndicator(),
+        if (snapshot.hasError) {
+          return _buildDayContainer();
+        } else if (snapshot.hasData) {
+          Workout? workout;
+          try {
+            workout = Workout.fromJsonQuerySnapshot(snapshot.data!);
+          } catch (_) {
+            /// This results in bad state, due to parsing error
+            // Log.error(e.toString(), stackTrace: stackTrace);
+          }
+          return GestureDetector(
+            onTap: () {
+              if (workout != null)
+                _navigateToPreviousWorkoutScreen(
+                  context,
+                  workout,
+                );
```

Comment:
would pull all this into a seperate method and call it here so we don't define methods inside the ui, or just add the workout not null check to the existing _navigateToPreviousWorkoutScreen method.